pkg/frontend/openshiftclustercredentials_post_test.go

```diff
@@ -289,8 +314,13 @@ func TestPostOpenShiftClusterCredentials(t *testing.T) {
 				}
 
 			} else {
+				b, err := ioutil.ReadAll(resp.Body)
+				if err != nil {
+					t.Fatal(err)
+				}
+
 				cloudErr := &api.CloudError{StatusCode: resp.StatusCode}
-				err = json.NewDecoder(resp.Body).Decode(&cloudErr)
+				err = json.Unmarshal(b, &cloudErr)
```

Comment:
I think, we should modify all our error check in tests this way to catch things like https://github.com/Azure/ARO-RP/pull/124/files#r371757408. I'll submit a separate PR.

Reply:
There is similar potential issue in the `tt.wantError == ""` branch. Here and in other tests. I'll revert this change and create a separate PR for all of it (both errors and successes) across the repo.